### ISSUE TITLE
is the semantic loss right?

### ISSUE THREAD

**Issue by codingwolfman** [luvf/MSTN#1]
#center loss more tricky
    s_c, t_c = update_centers(model, s_gen, t_gen, s_true_hot, t_clf, args)
    

    S_loss = center_loss(t_c, s_c)
    model.s_center = s_c.detach()
    model.t_center = t_c.detach()

by this way, only the gradient of current_centroid will be backward, because the global centroid is detached, if this, we can compute semantic loss just using MSELoss(current_s_centroid, current_t_centroid), i'm not sure

**Comment by luvf** [comment]
the detach is for :
1) to avoid a recurent structure => comutational cost increase at each batch
2) and keeping a recurent structure wouldn't have much sense since the loss and the gradient descend have already been applyed once.
